manifests/l/LibgenApps/LibgenDesktop/1.4.1/LibgenApps.LibgenDesktop.installer.yaml

```diff
@@ -1,23 +1,23 @@
-# Created using YamlCreate.ps1 v1.1.7
-# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
+# Created with YamlCreate.ps1 v2.0.4 $debug=QUSU.7-2-1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
 
 PackageIdentifier: LibgenApps.LibgenDesktop
 PackageVersion: 1.4.1
 MinimumOSVersion: 10.0.0.0
+InstallerType: wix
```

Comment:
what makes this a wix instead of a msi?

Reply:
![image](https://user-images.githubusercontent.com/56779163/150816421-7e6533e4-5c5d-4933-a9e8-d1d4a285ca53.png)

Reply:
Interesting, i have no idea how and what Wix is, do they come as .msi only or can it be .exe? or is it burn if its a executable?
I think we should look more into this InstallerType as it might explain some weirdness, however wingetcreate and yamlcreate needs to be updated.

Reply:
Busy man @Trenly

Reply:
@OfficialEsco YamlCreate already supports wix as an installer type. However, I'll make it so that it is like architecture where it asks the user to confirm the installer type it detected. 

We may need to look at better ways to get installer type than just extension matching

Reply:
> YamlCreate already supports wix as an installer type.

When does it look for the ProductCode? if it detects .msi? if so no problem then

> We may need to look at better ways to get installer type than just extension matching

We've had this discussion before and i think you said we needed some .NET dependency instead of pure Powershell(?)

Reply:
> When does it look for the ProductCode? if it detects .msi? if so no problem then
> 
Yes, but as Levvie mentioned over on my fork, we may need to switch to the COM API for fetching ProductCodes

> We've had this discussion before and i think you said we needed some .NET dependency instead of pure Powershell(?)
>
I thought that was for download progress. I'm sure there are ways to do it, and considering that I'm looking at the 3.0 refactor, I might try to build it in. It just depends on how complicated it gets

Reply:
```
                if ([System.Environment]::OSVersion.Platform -match 'Win') {
                    $MSIProductCode = [string]$(Get-AppLockerFileInformation -Path $script:dest | Select-Object Publisher | Select-String -Pattern '{[A-Z0-9]{8}-([A-Z0-9]{4}-){3}[A-Z0-9]{12}}').Matches
                }
                if (Test-String -not $MSIProductCode -IsNull) {
                    $_Installer['ProductCode'] = $MSIProductCode
                } elseif ( ($_Installer.Keys -contains 'ProductCode') -and ($_Installer.InstallerType -in @('appx'; 'msi'; 'msix'; 'appxbundle'; 'msixbundle'))) {
                    $_Installer.Remove('ProductCode')
                }
```
From my understanding it ALWAYS checks for a ProductCode if you're on Windows and removes it if its a appx, msi, msix, appxbundle or msixbundle

I went down a tiny rabbithole to see if we could get the get the info from the details page, but apparently that page is very inconsistent 🤔
For .msi it appears the best way is indeed COM API for any of the metadata
For .exe we could suggest some data from `(Get-Item 'C:\Users\User\Downloads\Firefox Setup 96.0.2.exe').VersionInfo | Format-List *` (FileVersionRaw, Language, CompanyName, FileDescription, LegalCopyright)

Reply:
Soooo uhhh yeah, we have a huuuuuuuuuuge InstallerType missmatch between MSI and WIX https://github.com/microsoft/winget-cli/issues/1857
Any installer that have `Program name: Windows Installer` is pure msi
Any installer that is `Program name: Windows Installer XML Toolset` is wix

Reply:
@Trenly YamlCreate needs a hotfix to clear the wix ProductCode

Reply:
> @Trenly YamlCreate 2.0.5 needs a hotfix to clear the wix ProductCode

I have it as part of using the COM API right now, but am having this issue. I'd like to switch to the COM API as part of this to ensure that we have the best data

https://github.com/Trenly/winget-pkgs/issues/138#issuecomment-1020421933

Reply:
@Trenly I have fixed your issue.

Reply:
Created PR for the fixes